### PR TITLE
Issue 75 make it possible for one person to have multiple censu

### LINKED ISSUE #75: Make it possible for one person to have multiple census features
# **Feature Summary**
Currently in #28 the function creates **one-hot encoding** - i.e. for census features `[f1, f2, f3]`, the function allocates a person a list of boolean `[True, False, False]` with the caveat that **there will always be one `True` in the list**. Extend the current algorithm to allow for **more than one `True` in the allocated census features**

(I'm not sure if this actually matters though..)

### PR DESCRIPTION
**Pull request by zazwaz12**
---
Title: "Closes #75 "
---

# **Summary of Changes**
A function that takes in a dictionary of a census grouping occurences and their values, then returns a lazy frame with the sample proportion for valid simulation.

# **Checklist:**
- [X] My code follows the code style of this project.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] New and existing unit tests prove my fix is effective or that my feature works.
- [X] I have annotated added types appropriately.

### **Additional context**
How this will work in will become more apparent once we have the classifications and categorization decided.

### CODE REVIEW

**Review by ctmes — APPROVED** [review]

File: nhs/data/allocation.py [hunk 1]
```diff
@@ -74,6 +74,39 @@ def join_census_with_coords(
     )
 
 
+@log_entry_exit()
+def calculate_proportions(group_values_dict: dict) -> pl.LazyFrame:
+
```
Comment (ctmes) [inline nhs/data/allocation.py:79]: maybe a comment about how it works and why

File: nhs/data/allocation.py [hunk 2]
```diff
@@ -74,6 +74,39 @@ def join_census_with_coords(
...
+    unique_dict = {}
+
+    for group, value in group_values_dict.items():
+        if group not in unique_dict:
+            unique_dict[group] = value
+
+    # Construct a LazyFrame from the unique dictionary
+    lf = pl.LazyFrame(
+        {"group_col": list(unique_dict.keys()), "value_col": list(unique_dict.values())}
+    )
```
Comment (SodaVolcano) [inline nhs/data/allocation.py:93]: Could you do this without making another lazy frame? so append a new column onto the existing dataframe for example

File: nhs/data/allocation.py [hunk 3]
```diff
@@ -74,6 +74,39 @@ def join_census_with_coords(
     )
 
 
+@log_entry_exit()
+def calculate_proportions(group_values_dict: dict) -> pl.LazyFrame:
```
Comment (SodaVolcano) [inline nhs/data/allocation.py:78]: input should be a lazyframe as well and not a dictionary

File: nhs/data/allocation.py [hunk 4]
```diff
@@ -74,6 +74,39 @@ def join_census_with_coords(
...
+        lf.select(pl.col("value_col").sum()).collect().get_column("value_col")[0]
+    )
+
+    # Add a new column for the proportions using a LazyFrame
+    lf = lf.with_columns(
+        (pl.col("value_col") / total_sum_expr).round(3).alias("proportion")
+    )
+
+    # Collect to evaluate the LazyFrame (for demonstration purposes)
+    print(lf.collect())
```
Comment (SodaVolcano) [inline nhs/data/allocation.py:106]: you can use `logger.debug()` instead of print

**Review by SodaVolcano — CHANGES_REQUESTED** [review]
Yup good progress, though I'll ask you to finish #75 before making a PR since this PR seems to be unfinished

**Comment by SodaVolcano** [comment]
See review comment above - try making a PR after adding the full feature for #75 , so far this PR only have proportion calculation

**Pull request closed without merging**